javaslang/src/main/java/javaslang/collection/CharSeq.java

```diff
@@ -2161,8 +2161,12 @@ public CharSeq replace(CharSequence target, CharSequence replacement) {
      * @throws PatternSyntaxException if the regular expression's syntax is invalid
      * @see java.util.regex.Pattern
      */
-    public Seq<CharSeq> split(String regex, int limit) {
-        Array<String> split = Array.wrap(back.split(regex, limit));
+    @Deprecated(/* Use splitSeq instead, will be removed in 3.0.0 */)
```

Comment:
this is the actual fix, replaced the impl with the new one (added back old tests also)